packages/nix/mkDevShells/pf-env-bootstrap-restore.sh

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env bash
+
+# This script restores the environments EKS cluster nodes and resources to their original state
+
+set -eo pipefail
+export AWS_PAGER=""
+
+REPO_VARIABLES=$(pf-get-repo-variables)
+KUBE_DIR=$(echo "$REPO_VARIABLES" | jq -r '.kube_dir')
+
+KUBE_CONFIG_PATH="$KUBE_DIR/config"
+STATE_FILE="$KUBE_DIR/cluster_suspend_state.json"
+
+####################################################################
+# Step 0: Validation
+####################################################################
+
+if [[ ! -f $KUBE_CONFIG_PATH ]]; then
+  echo "Error: No kubeconfig file found at $KUBE_CONFIG_PATH. Create it with 'pf-update-kube' first!" >&2
+  exit 1
+fi
+
+# Get the state file
+if [[ ! -f $STATE_FILE ]]; then
+  echo "Error: No state file found at $STATE_FILE. Exiting." >&2
+  exit 1
+fi
+
+####################################################################
+# Step 1: Select the cluster
+####################################################################
+
+EKS_CLUSTER_NAME=$(kubectl config get-contexts -o name | fzf --prompt="Select a Kubernetes cluster: ")
+CONTEXT_ARGS="--context=$EKS_CLUSTER_NAME"
+
+####################################################################
+# Step 2: Confirmation
+####################################################################
+
+echo -e "You selected \n" >&2
+echo -e "Cluster: $EKS_CLUSTER_NAME\n" >&2
+read -rp "Enter name of cluster to confirm: " CONFIRM_CLUSTER
+
+if [ "$EKS_CLUSTER_NAME" != "$CONFIRM_CLUSTER" ]; then
+  echo -e "$CONFIRM_CLUSTER does not match $EKS_CLUSTER_NAME Exiting.\n" >&2
+  exit 1
+fi
+
+####################################################################
+# Step 3: Retreive Restore Values
+####################################################################
+
+function get_cluster_state() {
+  local property_name=$1
+
+  if [[ ! -f $STATE_FILE ]]; then
+    echo "State file not found."
+    return 1
+  fi
+
+  # Retrieve the property value using jq
+  jq -r --arg cluster "$EKS_CLUSTER_NAME" --arg property_name "$property_name" \
+    '.[$cluster][$property_name] // "Property not found."' "$STATE_FILE"
+}
+
+aws_profile=$(get_cluster_state "aws_profile")
+export AWS_PROFILE=$aws_profile
+
+if [[ -z $AWS_PROFILE ]]; then
+  echo "Error: AWS profile not found in state file. Exiting." >&2
+  exit 1
+fi
+
+scale_asgs_nat_nodes() {
+  local prefix="nat-"
+  local min_size=$1
+  local desired_capacity=$2
+
+  asgs=$(aws autoscaling describe-auto-scaling-groups | jq -r ".AutoScalingGroups[] | select(.AutoScalingGroupName | startswith(\"$prefix\")) | .AutoScalingGroupName")
```

Comment:
This should be limited to just the NAT nodes that are in the same VPC as the EKS cluster.